src/lib/containers/download_list_v2/DownloadListTable.tsx

```diff
@@ -68,19 +70,33 @@ export default function DownloadListTable(props: DownloadListTableProps) {
     }
   }, [isError, newError, handleError])
 
+  const allRows = data?.pages.flatMap(page => page.page) ?? []
+
   useEffect(() => {
+    const copyAllSynapseIDs = () => {
+      const synIDs = allRows.map((item: DownloadListItemResult)=>{
+        return item.fileEntityId
+      }).join('\n')
+      // https://caniuse.com/mdn-api_clipboard_writetext
+      navigator.clipboard.writeText(synIDs).then(() => { 
+        displayToast('Successfully copied to clipboard')
+      })
+      setCopyingAllSynapseIDs(false)
+    }
+
     if (
       status === 'success' &&
       !isFetchingNextPage &&
       hasNextPage &&
       fetchNextPage &&
-      inView
+      (inView || copyingAllSynapseIDs)
     ) {
       fetchNextPage()
+    } else if (!hasNextPage && copyingAllSynapseIDs) {
```

Comment:
Step 3:  If we have all pages of data, and we're coyingAllSynapseIDs, then concat all of the Synapse IDs from the data and copy them to the clipboard.